sdk/src/lib.rs

```diff
@@ -347,30 +350,15 @@ impl WorkflowHalf {
     }
 }
 
-tokio::task_local! {
-    // This works, but maybe just passing a context object for activities like WFs is better
-    static ACT_CANCEL_TOK: CancellationToken
-}
-
-/// Returns a future the completes if and when the activity this was called inside has been
-/// cancelled
-pub async fn act_cancelled() {
-    ACT_CANCEL_TOK.with(|ct| ct.clone()).cancelled().await
-}
-
-/// Returns true if this activity has already been cancelled
-pub fn act_is_cancelled() -> bool {
-    ACT_CANCEL_TOK.with(|ct| ct.is_cancelled())
-}
-
 impl ActivityHalf {
     /// Spawns off a task to handle the provided activity task
     fn activity_task_handler(
         &mut self,
         worker: Arc<dyn CoreWorker>,
+        task_queue: String,
         activity: ActivityTask,
     ) -> Result<(), anyhow::Error> {
-        match activity.variant {
+        match activity.variant.clone() {
```

Comment:
We should avoid this clone - cloning the payloads can be potentially quite expensive.

It loooks like all we really need to to is take out the task token earlier (and it's fine to clone that).